features/machine/autoscaler.feature

```diff
@@ -52,3 +52,134 @@ Feature: Cluster Autoscaler Tests
     the expression should be true> machine_set.desired_replicas(cached: false) == 1
     """
     Then the machineset should have expected number of running machines
+  
+  # @author zhsun@redhat.com
+  # @case_id OCP-21516
+  @admin
+  @destructive
+  Scenario: Cao listens and deploys cluster-autoscaler based on ClusterAutoscaler resource
+    Given I have an IPI deployment
+    And I switch to cluster admin pseudo user
+
+    And admin ensures "default" clusterautoscaler is deleted after scenario
+    When I run the :create admin command with:
+      | f | https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/cloud/cluster-autoscaler.yml   |
+    Then the step should succeed
+    And 1 pods become ready with labels:
+      | cluster-autoscaler=default,k8s-app=cluster-autoscaler                                                 |
+
+  # @author zhsun@redhat.com
+  # @case_id OCP-21517
+  @admin
+  @destructive
+  Scenario: CAO listens and annotations machineSets based on MachineAutoscaler resource
+    Given I have an IPI deployment
+    And I switch to cluster admin pseudo user
+
+    Given I store the number of machines in the :num_to_restore clipboard
+    And admin ensures node number is restored to "<%= cb.num_to_restore %>" after scenario
+    And admin ensures "machineset-clone" machineset is deleted after scenario
+    And admin ensures "maotest" machineautoscaler is deleted after scenario
```

Comment:
Please put each `admin ensures delete` right after their corresponding creation. This is for their read consistency.

For example:

```
Given I clone a machineset named "machineset-clone"
And admin ensures "machineset-clone" machineset is deleted after scenario
```